### ISSUE TITLE
Have some process for auto-generating documentation

### ISSUE THREAD

**Issue by mgrauer** [girder/girder_web_components#14]
xref https://github.com/girder/girder/pull/2563.

**Comment by jeffbaumes** [comment]
This looks interesting

https://github.com/vue-styleguidist/vue-styleguidist

Since this is a component library, it would seem that the focus should be on great component docs.

There is also the Girder REST library and utils, which could perhaps generate a second doc site for this library with more standard JSDoc/ESDoc (with a main page pointing to "Components" and "API" docs), or it could be spun off into its own library, since this is the "components" repo.

**Comment by jbeezley** [comment]
I spent some time looking into this for resonantgeo.  Many different options exist, but styleguidist seemed the most promising as a vue native tool.  I didn't use it at the time because it didn't support pug, but it looks like it has been [added](https://github.com/vue-styleguidist/vue-docgen-api/pull/49) recently.

**Comment by zachmullen** [comment]
I just tried to use vue-styleguidist via its vue-cli plugin, but ran into [an issue](https://github.com/vue-styleguidist/vue-cli-plugin-styleguidist/issues/17) that I couldn't resolve, even in a barebones minimal reproduction. Hopefully we'll hear back on that, until then we can't document anything that uses `async/await`.

**Comment by subdavis** [comment]
Worth looking into again, maybe.

**Comment by zachmullen** [comment]
Good call. I don't know whether I made an earnest attempt on GWC after that upstream fix.
